# 04 · Target and temporal split
Why not random split: the same customer appears at many T0s; a random split would train on the future (SPEC §6).

In [1]:
import os, pathlib
os.chdir(pathlib.Path.cwd().parent if pathlib.Path.cwd().name == "notebooks" else pathlib.Path.cwd())
import pandas as pd, numpy as np, matplotlib.pyplot as plt
from IPython.display import Image, display
pd.set_option("display.width", 200); pd.set_option("display.max_columns", 30)
from retail_targeting.config import load_config
cfg = load_config()
T = lambda name: pd.read_csv(f"outputs/tables/{name}")

In [2]:
from retail_targeting.config import split_dates
split_dates(cfg)

{'train': [datetime.date(2010, 6, 1),
  datetime.date(2010, 7, 1),
  datetime.date(2010, 8, 1),
  datetime.date(2010, 9, 1),
  datetime.date(2010, 10, 1),
  datetime.date(2010, 11, 1),
  datetime.date(2010, 12, 1),
  datetime.date(2011, 1, 1)],
 'validation': [datetime.date(2011, 4, 1), datetime.date(2011, 5, 1)],
 'test': [datetime.date(2011, 8, 1), datetime.date(2011, 9, 1)]}

In [3]:
s = T('snapshot_summary.csv'); s.pivot_table(index='decision_date', columns='split', values='prevalence')

split,purged,test,train,validation
decision_date,,,,
2010-06-01,NaN,NaN,0.500377,NaN
2010-07-01,NaN,NaN,0.509317,NaN
2010-08-01,NaN,NaN,0.570210,NaN
2010-09-01,NaN,NaN,0.620128,NaN
2010-10-01,NaN,NaN,0.579510,NaN
2010-11-01,NaN,NaN,0.493429,NaN
2010-12-01,NaN,NaN,0.380375,NaN
2011-01-01,NaN,NaN,0.377314,NaN
2011-02-01,0.396655,NaN,NaN,NaN


Prevalence varies 0.38–0.62 with season: T0s whose outcome window covers Q4 have higher repeat rates. Test (T0 = Aug/Sep 2011) has prevalence 0.58 vs ~0.50 in train.